File: 004-anagramdetector.rb
require "pry"

def anagramsorter(stringpair)
	firstword = stringpair[0].gsub(/ /, "")
	secondword = stringpair[1].gsub(/ /, "")
	i =0
	check = true
	while i < firstword.length do
		while check == true do
			character = firstword[i]
			if character == nil
				check = false
			else
				check = secondword.include? character
				secondword = secondword.gsub(character, "")
				binding.pry
				i += 1
			end
		end
	end
	return secondword
end

def anagramchecker(secondwordwhatsleft)

	if secondwordwhatsleft == ""
		return true
	else
		return false
	end
end

input1 = ["wisdom", "mid sow"]
input2 = ["Seth Rogan", "Gathers No"]
#input3 = ["Reddit", "Eat Dirt"]
#input4 = ["Schoolmaster", "The classroom"]
#input5 = ["Astronomers", "Moon starer"]
# input6 = ["Vacation Times", "I'm Not as Active"]


#firstresult1 = anagramsorter(input1)
#result1 = anagramchecker(firstresult1)


result2 = anagramchecker(anagramsorter(input2))
#result3 = anagramchecker(anagramsorter(input3))
#result4 = anagramchecker(anagramsorter(input4))
#result5 = anagramchecker(anagramsorter(input5))
# result6 = anagramchecker(anagramsorter(input6))

#puts result1
puts result2
#puts result3
#puts result4
#puts result5
# puts result6

File: 004-anagramdetector.js
function anagramsorter(stringpair){
	var firstword = stringpair[0].replace(/ /g, "");
	var secondword = stringpair[1].replace(/ /g, "");

	for (var i = 0; i < firstword.length; i++){
		while (check == true) {
			var character = firstword(i);
			var check = secondword.includes(character);
			secondword = secondword.replace(character, "");
		}
	}
	return secondword;
}

function anagramchecker(secondwordwhatsleft){

	if (secondwordwhatsleft == ""){
		return true;
	}
	else
		return false;
	
}

var input1 = ["wisdom", "mid sow"];
var input2 = ["Seth Rogan", "Gathers No"];
var input3 = ["Reddit", "Eat Dirt"];
var input4 = ["Schoolmaster", "The classroom"];
var input5 = ["Astronomers", "Moon starer"];
var input6 = ["Vacation Times", "I'm Not as Active"];

var result1 = anagramchecker(anagramsorter(input1));
var result2 = anagramchecker(anagramsorter(input2));
var result3 = anagramchecker(anagramsorter(input3));
var result4 = anagramchecker(anagramsorter(input4));
var result5 = anagramchecker(anagramsorter(input5));
var result6 = anagramchecker(anagramsorter(input6));

console.log(result1);
console.log(result2);
console.log(result3);
console.log(result4);
console.log(result5);
console.log(result6);